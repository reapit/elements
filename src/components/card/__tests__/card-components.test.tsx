import { render } from '@testing-library/react'
import { Card, handleMouseHover } from '../card-components'

describe('Card', () => {
  it('should match a snapshot and render children with no props', () => {
    const wrapper = render(
      <Card>
        <div>I am a child</div>
      </Card>,
    )
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should match a snapshot and render children with full props', () => {
    const wrapper = render(
      <Card
        hasMainCard
        hasListCard
        mainCardHeading="Main Heading"
        mainCardSubHeading="Main Subheading"
        mainCardSubHeadingAdditional="Main Subheading Additional"
        mainCardBody="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
        mainCardImgUrl="https://via.placeholder.com/72x72.svg"
        listCardHeading="List Card Heading"
        listCardSubHeading="List Card Sub Heading"
        listCardItems={[
          {
            listCardItemHeading: 'Applicant',
            listCardItemSubHeading: 'Bob Smith',
            listCardItemIcon: 'user',
            onClick: () => console.log('Clicking'),
          },
          {
            listCardItemHeading: 'Property',
            listCardItemSubHeading: 'Some Address',
            listCardItemIcon: 'property',
            onClick: () => console.log('Clicking'),
          },
        ]}
      >
        <div>I am a child</div>
      </Card>,
    )
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})

describe('handleMouseHover', () => {
  it('should set the hover index', () => {
    const hoverIndex = 1
    const setHoverIndex = vi.fn()

    const curried = handleMouseHover(hoverIndex, setHoverIndex)

    curried()

    expect(setHoverIndex).toHaveBeenCalledWith(hoverIndex)
  })
})
