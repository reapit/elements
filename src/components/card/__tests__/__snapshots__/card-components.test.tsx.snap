// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > should match a snapshot and render children with full props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mocked-styled-0 el-card-wrap"
      >
        <div
          class="mocked-styled-2 el-card-main-wrap"
        >
          <div
            class="mocked-styled-20 el-deprecated-avatar el-deprecated-avatar-image el-deprecated-avatar"
            role="img"
            title="An avatar image"
          >
            <img
              alt="An image with source https://via.placeholder.com/72x72.svg"
              src="https://via.placeholder.com/72x72.svg"
            />
          </div>
          <div
            class="mocked-styled-1 el-card-heading-wrap"
          >
            <h5
              class="mocked-styled-3 el-card-heading"
            >
              Main Heading
            </h5>
            <h6
              class="mocked-styled-4 el-card-sub-heading"
            >
              Main Subheading
            </h6>
            <h6
              class="mocked-styled-5 el-card-sub-heading-additional"
            >
              Main Subheading Additional
            </h6>
          </div>
        </div>
        <div
          class="el-mb5 el-card-body-wrap"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
        </div>
        <div
          class="el-mt5 el-card-list-main-wrap"
        >
          <h5
            class="mocked-styled-11 el-card-list-heading"
          >
            List Card Heading
          </h5>
          <h6
            class="mocked-styled-12 el-card-list-sub-heading"
          >
            List Card Sub Heading
          </h6>
        </div>
        <div
          class="mocked-styled-13 el-card-list-item"
        >
          <div
            class="mocked-styled-17 el-card-list-icon"
          >
            <span
              class="el-intent-primary el-icon"
            >
              <svg
                fill="none"
                height="24"
                role="img"
                title="Icon image with name user"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M8.37998 8.88C8.37998 6.87911 9.99911 5.25551 12 5.25C14.0025 5.25549 15.6245 6.87749 15.63 8.88C15.6272 9.49712 15.5703 10.1128 15.46 10.72C15.3083 12.534 13.8195 13.9453 12 14C10.1805 13.9453 8.69171 12.534 8.54 10.72C8.43267 10.1126 8.37913 9.49685 8.37998 8.88ZM12 12C12.91 12 13.36 11.18 13.49 10.4C13.5796 9.89821 13.6265 9.38972 13.63 8.88C13.678 8.26901 13.3792 7.68276 12.8566 7.36257C12.334 7.04238 11.676 7.04238 11.1534 7.36257C10.6308 7.68276 10.332 8.26901 10.38 8.88C10.3789 9.38949 10.4224 9.8981 10.51 10.4C10.64 11.18 11.09 12 12 12Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M1 12C1 5.92487 5.92487 1 12 1C14.9174 1 17.7153 2.15893 19.7782 4.22183C21.8411 6.28473 23 9.08262 23 12C23 18.0751 18.0751 23 12 23C5.92487 23 1 18.0751 1 12ZM12 17C9.25 17 7.27 18.27 7 19.49H7.06C10.0671 21.5044 13.9929 21.5044 17 19.49C16.73 18.27 14.75 17 12 17ZM12 15C15.07 15 17.62 16.28 18.55 18.12H18.58C21.9161 14.545 21.7874 8.9604 18.2902 5.54284C14.793 2.12528 9.20695 2.12528 5.70975 5.54284C2.21255 8.9604 2.0839 14.545 5.42 18.12C6.38 16.28 8.93 15 12 15Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </div>
          <div
            class="mocked-styled-14 el-card-list-item-text-wrap"
          >
            <div
              class="mocked-styled-15 el-card-list-item-text-primary"
            >
              Applicant
            </div>
            <div
              class="mocked-styled-16 el-card-list-item-text-secondary"
            >
              Bob Smith
            </div>
          </div>
        </div>
        <div
          class="mocked-styled-13 el-card-list-item"
        >
          <div
            class="mocked-styled-17 el-card-list-icon"
          >
            <span
              class="el-intent-primary el-icon"
            >
              <svg
                fill="none"
                height="24"
                role="img"
                title="Icon image with name property"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M22.9227 9.16484L12.4227 2.16484C12.089 1.94505 11.6565 1.94505 11.3227 2.16484L0.872748 9.16484C0.470624 9.48875 0.381235 10.0656 0.66646 10.496C0.951685 10.9264 1.51772 11.0689 1.97275 10.8248L2.87275 10.1948V21.4948C2.87275 22.0471 3.32046 22.4948 3.87275 22.4948H19.8727C20.425 22.4948 20.8727 22.0471 20.8727 21.4948V10.1948L21.8227 10.8248C21.9838 10.9376 22.1761 10.9971 22.3727 10.9948C22.8113 10.9914 23.1965 10.7027 23.3227 10.2827C23.4489 9.86268 23.2868 9.40943 22.9227 9.16484ZM18.8727 20.4948H4.87275V8.86484L11.8727 4.19484L18.8727 8.86484V20.4948Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
          <div
            class="mocked-styled-14 el-card-list-item-text-wrap"
          >
            <div
              class="mocked-styled-15 el-card-list-item-text-primary"
            >
              Property
            </div>
            <div
              class="mocked-styled-16 el-card-list-item-text-secondary"
            >
              Some Address
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mocked-styled-0 el-card-wrap"
    >
      <div
        class="mocked-styled-2 el-card-main-wrap"
      >
        <div
          class="mocked-styled-20 el-deprecated-avatar el-deprecated-avatar-image el-deprecated-avatar"
          role="img"
          title="An avatar image"
        >
          <img
            alt="An image with source https://via.placeholder.com/72x72.svg"
            src="https://via.placeholder.com/72x72.svg"
          />
        </div>
        <div
          class="mocked-styled-1 el-card-heading-wrap"
        >
          <h5
            class="mocked-styled-3 el-card-heading"
          >
            Main Heading
          </h5>
          <h6
            class="mocked-styled-4 el-card-sub-heading"
          >
            Main Subheading
          </h6>
          <h6
            class="mocked-styled-5 el-card-sub-heading-additional"
          >
            Main Subheading Additional
          </h6>
        </div>
      </div>
      <div
        class="el-mb5 el-card-body-wrap"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
      </div>
      <div
        class="el-mt5 el-card-list-main-wrap"
      >
        <h5
          class="mocked-styled-11 el-card-list-heading"
        >
          List Card Heading
        </h5>
        <h6
          class="mocked-styled-12 el-card-list-sub-heading"
        >
          List Card Sub Heading
        </h6>
      </div>
      <div
        class="mocked-styled-13 el-card-list-item"
      >
        <div
          class="mocked-styled-17 el-card-list-icon"
        >
          <span
            class="el-intent-primary el-icon"
          >
            <svg
              fill="none"
              height="24"
              role="img"
              title="Icon image with name user"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M8.37998 8.88C8.37998 6.87911 9.99911 5.25551 12 5.25C14.0025 5.25549 15.6245 6.87749 15.63 8.88C15.6272 9.49712 15.5703 10.1128 15.46 10.72C15.3083 12.534 13.8195 13.9453 12 14C10.1805 13.9453 8.69171 12.534 8.54 10.72C8.43267 10.1126 8.37913 9.49685 8.37998 8.88ZM12 12C12.91 12 13.36 11.18 13.49 10.4C13.5796 9.89821 13.6265 9.38972 13.63 8.88C13.678 8.26901 13.3792 7.68276 12.8566 7.36257C12.334 7.04238 11.676 7.04238 11.1534 7.36257C10.6308 7.68276 10.332 8.26901 10.38 8.88C10.3789 9.38949 10.4224 9.8981 10.51 10.4C10.64 11.18 11.09 12 12 12Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M1 12C1 5.92487 5.92487 1 12 1C14.9174 1 17.7153 2.15893 19.7782 4.22183C21.8411 6.28473 23 9.08262 23 12C23 18.0751 18.0751 23 12 23C5.92487 23 1 18.0751 1 12ZM12 17C9.25 17 7.27 18.27 7 19.49H7.06C10.0671 21.5044 13.9929 21.5044 17 19.49C16.73 18.27 14.75 17 12 17ZM12 15C15.07 15 17.62 16.28 18.55 18.12H18.58C21.9161 14.545 21.7874 8.9604 18.2902 5.54284C14.793 2.12528 9.20695 2.12528 5.70975 5.54284C2.21255 8.9604 2.0839 14.545 5.42 18.12C6.38 16.28 8.93 15 12 15Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
        <div
          class="mocked-styled-14 el-card-list-item-text-wrap"
        >
          <div
            class="mocked-styled-15 el-card-list-item-text-primary"
          >
            Applicant
          </div>
          <div
            class="mocked-styled-16 el-card-list-item-text-secondary"
          >
            Bob Smith
          </div>
        </div>
      </div>
      <div
        class="mocked-styled-13 el-card-list-item"
      >
        <div
          class="mocked-styled-17 el-card-list-icon"
        >
          <span
            class="el-intent-primary el-icon"
          >
            <svg
              fill="none"
              height="24"
              role="img"
              title="Icon image with name property"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M22.9227 9.16484L12.4227 2.16484C12.089 1.94505 11.6565 1.94505 11.3227 2.16484L0.872748 9.16484C0.470624 9.48875 0.381235 10.0656 0.66646 10.496C0.951685 10.9264 1.51772 11.0689 1.97275 10.8248L2.87275 10.1948V21.4948C2.87275 22.0471 3.32046 22.4948 3.87275 22.4948H19.8727C20.425 22.4948 20.8727 22.0471 20.8727 21.4948V10.1948L21.8227 10.8248C21.9838 10.9376 22.1761 10.9971 22.3727 10.9948C22.8113 10.9914 23.1965 10.7027 23.3227 10.2827C23.4489 9.86268 23.2868 9.40943 22.9227 9.16484ZM18.8727 20.4948H4.87275V8.86484L11.8727 4.19484L18.8727 8.86484V20.4948Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="mocked-styled-14 el-card-list-item-text-wrap"
        >
          <div
            class="mocked-styled-15 el-card-list-item-text-primary"
          >
            Property
          </div>
          <div
            class="mocked-styled-16 el-card-list-item-text-secondary"
          >
            Some Address
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card > should match a snapshot and render children with no props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mocked-styled-0 el-card-wrap"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="mocked-styled-0 el-card-wrap"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
